client/src/pages/locations/Form.js

```diff
@@ -138,16 +140,38 @@ const BaseLocationForm = ({ currentUser, edit, title, initialValues }) => {
                   name="location"
                   component={FieldHelper.ReadonlyField}
                   humanValue={
-                    <>
-                      <Coordinate coord={values.lat} />,{" "}
-                      <Coordinate coord={values.lng} />
-                    </>
+                    <GeoLocation
+                      lat={values.lat}
+                      lng={values.lng}
+                      isSubmitting={isSubmitting}
+                      onClearLocation={() => onClearLocation(setFieldValue)}
+                    />
                   }
                 />
+
+                <Field
+                  name="lat"
+                  component={FieldHelper.InputField}
+                  onBlur={() => {
+                    setFieldTouched("lat", true, false)
+                    setFieldValue("lat", Location.parseCoordinate(values.lat))
+                    updateMarkers = !updateMarkers
```

Comment:
Do not use `updateMarkers` as it is a way to explicitly force an update on the map which should be detected by the change in the `markers`